crates/core/args.rs

```diff
@@ -576,60 +576,87 @@ impl ArgMatches {
     ///
     /// If there was a problem building the matcher (e.g., a syntax error),
     /// then this returns an error.
-    #[cfg(feature = "pcre2")]
     fn matcher(&self, patterns: &[String]) -> Result<PatternMatcher> {
         if self.is_present("pcre2") {
-            let matcher = self.matcher_pcre2(patterns)?;
-            Ok(PatternMatcher::PCRE2(matcher))
+            self.matcher_engine("pcre2", patterns)
         } else if self.is_present("auto-hybrid-regex") {
-            let rust_err = match self.matcher_rust(patterns) {
-                Ok(matcher) => return Ok(PatternMatcher::RustRegex(matcher)),
-                Err(err) => err,
-            };
-            log::debug!(
-                "error building Rust regex in hybrid mode:\n{}",
-                rust_err,
-            );
-            let pcre_err = match self.matcher_pcre2(patterns) {
-                Ok(matcher) => return Ok(PatternMatcher::PCRE2(matcher)),
-                Err(err) => err,
-            };
-            Err(From::from(format!(
-                "regex could not be compiled with either the default regex \
-                 engine or with PCRE2.\n\n\
-                 default regex engine error:\n{}\n{}\n{}\n\n\
-                 PCRE2 regex engine error:\n{}",
-                "~".repeat(79),
-                rust_err,
-                "~".repeat(79),
-                pcre_err,
-            )))
+            self.matcher_engine("auto-hybrid", patterns)
         } else {
-            let matcher = match self.matcher_rust(patterns) {
-                Ok(matcher) => matcher,
-                Err(err) => {
-                    return Err(From::from(suggest_pcre2(err.to_string())));
-                }
-            };
-            Ok(PatternMatcher::RustRegex(matcher))
+            let engine = self.value_of_lossy("engine")
+                             .ok_or(clap::Error::with_description("Please \
+                                     provide a valid value to the engine \
+                                     option.", clap::ErrorKind::InvalidValue))?;
```

Comment:
Could you create errors in a way that is consistent with the surrounding code? e.g., `From::from(format!("..."))`.

Reply:
Actually, now that I think about it, you should just be able to call `unwrap()` here since `engine` has a default value.